extensions/2.0/Vendor/EXT_primitive_voxels/README.md

```diff
@@ -49,34 +61,195 @@ This extension allows primitives to use their attribute data to represent volume
 }
 ```
 
-The extension adds three new primitive modes, corresponding to voxel grid geometries:
-- `0x80000000` (`2147483648`) - A Cartesian box. The grid is a Cartesian grid of equally-sized boxes.
-- `0x80000001` (`2147483649`) - A cylinder. The grid is a stack of concentric rings, evenly divided around the circumference.
-- `0x80000002` (`2147483650`) - An ellipsoid. The grid is a set of concentric ellipsoids, divided evenly in latitude and longitude.
-
-The lowest byte is reserved for future voxel modes: `0x80000000`-`0x800000FF`.
+Though voxels are commonly associated with cubic geometry on a box-based grid, this extension also allows voxels to be based on other shapes, including cylinder-based regions specified by [`EXT_implicit_cylinder_region`](../EXT_implicit_cylinder_region/) and ellipsoid-based regions specified by [`EXT_implicit_ellipsoid_region`](../EXT_implicit_ellipsoid_region/). The supported shapes are visualized below.
 
-|Box|Cylinder|Ellipsoid|
+|Box|Cylinder|Region|
 | ------------- | ------------- | ------------- |
-|![Rectangular Voxel Grid](figures/box.png)|![Cylindrical Voxel Grid](figures/cylinder.png)|![Ellipsoid Voxel Grid](figures/sphere.png)|
+|![Box Voxel Grid](figures/box.png)|![Cylindrical Voxel Grid](figures/cylinder.png)|![Ellipsoid Voxel Grid](figures/sphere.png)|
+
+Voxels exist inside a bounding volume that conforms to the shape of the grid. The `dimensions` property refers to the number of subdivisions _within_ this bounding volume. Each value of `dimensions` must be a positive integer.
+
+The relationship between `dimensions` and the grid geometry is explained below.
+
+### Box Grid
+
+A **box** grid is a Cartesian grid defined by `x`, `y`, and `z` axes with equally-sized boxes. The `dimensions` correspond to the subdivisions of the box along the `x`, `y`, and `z` axes respectively. Elements are laid out in memory where the `x` data is contiguous up to stride.
+
+![Uniform box grid](figures/uniform-box.png)
+<p align="center"><i>A uniform box grid that is subdivided into two cells along each axis.</i></p>
+
+![Non-uniform box grid](figures/non-uniform-box.png)
+<p align="center"><i>A box grid that is non-uniformly scaled and also non-uniformly subdivided.</i></p>
+
+### Cylinder Region Grid
+
+A **cylinder** region grid is subdivided along the radius, height, and angle ranges of the region, visualized below.
+
+![Cylinder subdivisions](figures/cylinder-subdivisions.png)
+
+The cylinder is aligned with the `y`-axis in the primitive's local space. Its height is subdivided along that local `y`-axis from bottom to top. Subdivisions along the radius are concentric, centered around the `y`-axis and extending outwards. Finally, the angular bounds subdivided clockwise around the circumference of the cylinder.
+
+Elements are laid out in memory where data along the radius is contiguous up to stride.
```

Comment:
I think I understand based on context, but could you clarify what "up to stride" means?

Reply:
Ah "up to stride" is an artifact from the old extension README. I wasn't entirely sure what it meant either 😅 but I can rewrite it in a way that's more clear